std/format.d

```diff
@@ -978,6 +978,18 @@ uint formattedRead(R, Char, S...)(ref R r, const(Char)[] fmt, auto ref S args)
     assert(aa3 == ["hello":1, "world":2]);
 }
 
+// test rvalue using
+@system pure unittest
+{
+    string[int] aa1;
+    formattedRead!("%s")(`[1:"hello", 2:"world"]`, &aa1);
+    assert(aa1 == [1:"hello", 2:"world"]);
+
+    int[string] aa2;
+    formattedRead(`{"hello"=1; "world"=2}`, "{%(%s=%s; %)}", &aa2);
```

Comment:
I see that we only have one public example here. What do you think about adding another public example which shows fancy stuff like reading an array or associative array?
(This is independent from this PR, of course)